internal/pkg/cli/env_delete.go

```
@@ -291,7 +301,26 @@ func (o *deleteEnvOpts) validateNoRunningServices() error {
 				svcNames = append(svcNames, *t.Value)
 			}
 		}
-		return fmt.Errorf("service '%s' still exist within the environment %s", strings.Join(svcNames, ", "), o.name)
+		return fmt.Errorf("service %q still exist within the environment %s", strings.Join(svcNames, ", "), o.name)
+	}
+	return nil
+}
+
+func (o *deleteEnvOpts) validateNoDependencyPipelines() error {
+	pipelines, err := o.deployedPipelineLister.ListDeployedPipelines(o.appName)
+	if err != nil {
+		return fmt.Errorf("list deployed pipelines: %w", err)
+	}
+	for _, pipeline := range pipelines {
+		info, err := o.pipelineGetter.GetPipeline(pipeline.ResourceName)
+		if err != nil {
+			return fmt.Errorf("get pipeline %s: %w", pipeline.ResourceName, err)
+		}
+		for _, stage := range info.Stages {
+			if strings.TrimPrefix(stage.Name, deploy.StageFullNamePrefix) == o.name {
+				return fmt.Errorf("pipeline %q still uses the environment %s", pipeline.Name, o.name)
```

```suggestion
				return fmt.Errorf("pipeline %q depends on environment %s", pipeline.Name, o.name)
```

---

maybe we should return an error with `RecommendActions()` and say to remove the env dependency first, and run `env delete` again?